apps/prairielearn/src/pages/shared/syncHelpers.js

```diff
@@ -53,48 +56,57 @@ export async function gitStatus(locals) {
   return serverJob.jobSequenceId;
 }
 
-function locateImage(image, callback) {
-  docker.listImages(function (err, list = []) {
-    if (ERR(err, callback)) return;
-    for (let i = 0, len = list.length; i < len; i++) {
-      if (list[i].RepoTags && list[i].RepoTags?.indexOf(image) !== -1) {
-        return callback(null, docker.getImage(list[i].Id));
-      }
-    }
-    return callback(new Error(`Unable to find image=${image}`));
-  });
-}
-
-function confirmOrCreateECRRepo(repo, job, callback) {
+/**
+ * @param {string} repo
+ * @param {import('../../lib/server-jobs').ServerJob} job
+ */
+async function ensureECRRepo(repo, job) {
   const ecr = new ECR(makeAwsClientConfig());
   job.info(`Describing repositories with name: ${repo}`);
-  ecr.describeRepositories({ repositoryNames: [repo] }, (err, data) => {
-    let repositoryFound = false;
-    if (err) {
-      job.info(`Error returned from describeRepositories(): ${err}`);
-      job.info('Treating this error as meaning the desired repository does not exist');
+
+  let repositoryFound = false;
+  try {
+    const data = await ecr.describeRepositories({ repositoryNames: [repo] });
+    repositoryFound = !!data.repositories?.find((r) => r.repositoryName === repo);
+  } catch (err) {
+    if (err instanceof RepositoryNotFoundException) {
+      // Repository not found; this is expected.
     } else {
-      repositoryFound = !!_.find(data.repositories, ['repositoryName', repo]);
+      // Something else went wrong; allow it to bubble up.
+      Sentry.captureException(err, {
+        tags: {
+          repository: repo,
+        },
+      });
+      throw err;
     }
+  }
 
-    if (!repositoryFound) {
-      job.info('Repository not found');
-
-      job.info(`Creating repository: ${repo}`);
-      const params = {
-        repositoryName: repo,
-      };
-      ecr.createRepository(params, (err) => {
-        if (ERR(err, callback)) return;
-        job.info('Successfully created repository');
-        callback(null);
-      });
-    } else {
-      job.info('Repository found');
-      // Already exists, nothing to do
-      callback(null);
+  if (!repositoryFound) {
+    job.info('Repository not found');
+
+    job.info(`Creating repository: ${repo}`);
+    try {
+      await ecr.createRepository({ repositoryName: repo });
+    } catch (err) {
+      if (err instanceof RepositoryAlreadyExistsException) {
+        // Someone else created the repository before we could; this is fine.
+      } else {
+        // Something else went wrong; allow it to bubble up.
+        Sentry.captureException(err, {
+          tags: {
+            repository: repo,
+          },
+        });
+        throw err;
+      }
     }
-  });
+
+    job.info('Successfully created repository');
+  } else {
+    // Already exists, nothing to do
+    job.info('Repository found');
+  }
```

Comment:
Nit: instead of having a long if with a short else, I think the code could be made simpler if the condition is `if (repositoryFound)` and then call `job.info` and return, so you don't need the else. This would also reduce a level of indentation.

Reply:
Great idea, done in 84e2077.